### PR TITLE
[hotfix][docs] Re-introduce note about FLINK_CONF_DIR

### PR DESCRIPTION
**Pull request by rmetzger**
This PR addresses two smaller documentation concerns. The changes to the elastic scaling pages were suggested by @knaufk , the changes on the configuration page by @alpinegizmo and @xintongsong.

I'll adjust the Chinese variants of the docs once the changes have been approved.

### CODE REVIEW

File: docs/content/docs/deployment/config.md
Comment (alpinegizmo) [inline docs/content/docs/deployment/config.md:37]: What happens if I set FLINK_CONF_DIR when submitting a job to a session cluster?

```suggestion
You can specify a different configuration directory location by defining the `FLINK_CONF_DIR` environment variable. For resource providers which provide non-session deployments, you can specify per-job configurations this way. Make a copy of the `conf` directory from the Flink distribution and modify the settings on a per-job basis. Note that this is not supported in Docker or standalone Kubernetes deployments.
```

**Review by alpinegizmo — COMMENTED** [review]
Added some spelling suggestions.

It's not entirely clear to me what happens with FLINK_CONF_DIR on session clusters, but I guess it's ignored?

File: docs/content/docs/deployment/config.md
Comment (xintongsong) [inline docs/content/docs/deployment/config.md:38]: Maybe also mention `FLINK_PROPERTIES` as the alternative in docker/standalone-k8s deployments. This, instead of specifying a different configuration directory, appends given configurations to the flink-conf file in the image.
https://ci.apache.org/projects/flink/flink-docs-release-1.13/docs/deployment/resource-providers/standalone/docker/

File: docs/content/docs/deployment/config.md
Comment (rmetzger) [reply docs/content/docs/deployment/config.md:37]: thanks for the feedback. Looks like something is not correctly wired in my 🧠  😂

**Review by alpinegizmo — APPROVED** [review]
Thanks, Robert! LGTM.

**Comment by rmetzger** [comment]
Thanks a lot for the fast review. I'll wait for @knaufk's approval.

**Review by xintongsong — APPROVED** [review]
Thanks @rmetzger. Also looks good from my side.

**Review by knaufk — APPROVED** [review]
The elastic scaling changes look good to me. Thanks @rmetzger.

**Comment by rmetzger** [comment]
Thanks for the reviews! Merging ...

**Pull request closed without merging**
